xbmc/games/controllers/windows/GUIControllerWindow.cpp

```diff
@@ -189,10 +189,27 @@ void CGUIControllerWindow::OnEvent(const ADDON::CRepositoryUpdater::RepositoryUp
   UpdateButtons();
 }
 
+void CGUIControllerWindow::OnEvent(const ADDON::AddonEvent& event)
+{
+  using namespace ADDON;
+  
+  if (typeid(event) == typeid(AddonEvents::Enabled) ||  // also called on install,
+      typeid(event) == typeid(AddonEvents::Disabled) || // not called on uninstall
+      typeid(event) == typeid(AddonEvents::UnInstalled) ||
+      typeid(event) == typeid(AddonEvents::ReInstalled))
+  {
+    if (CServiceBroker::GetAddonMgr().HasType(event.id, ADDON_GAME_CONTROLLER))
+    {
+      UpdateButtons();
+    }
+  }
+}
+
 void CGUIControllerWindow::OnInitWindow(void)
 {
   // subscribe to events
```

Comment:
Can you move this block to the end of the function in a separate commit? I just noticed that we should subscribe to external components after initializing to prevent race conditions.